### PR TITLE
[Merged by Bors] - chore: Rename `DenseEmbedding` to `IsDenseEmbedding`

### PR DESCRIPTION
**Pull request by YaelDillies**
`Function.Embedding` is a type while `Embedding` is a proposition, and there are many other kinds of embeddings than topological embeddings. Hence this PR is a step towards
1. renaming `Embedding` to `IsEmbedding` and similarly for neighborhing declarations (which `DenseEmbedding` is)
2. namespacing it inside `Topology`

[Zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/rename.20.60Inducing.60.20and.20.60Embedding.60.3F). See #15993 for context.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by sgouezel** [comment]
Can you add deprecations?

File: Mathlib/Topology/DenseEmbedding.lean [hunk 1]
```diff
@@ -13,7 +13,7 @@ This file defines three properties of functions:
 
 * `DenseRange f`      means `f` has dense image;
 * `IsDenseInducing i` means `i` is also `Inducing`, namely it induces the topology on its codomain;
-* `DenseEmbedding e`  means `e` is further an `Embedding`, namely it is injective and `Inducing`.
+* `IsDenseEmbedding e`  means `e` is further an `Embedding`, namely it is injective and `Inducing`.
```
Comment (sgouezel) [inline Mathlib/Topology/DenseEmbedding.lean:16]:
```suggestion
* `IsDenseEmbedding e` means `e` is further an `Embedding`, namely it is injective and `Inducing`.
```
(and maybe adjust the whitespace in neighboring lines)?

**Review by sgouezel — COMMENTED** [review]
bors d+

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**